src/storage/Backend.cc

```diff
@@ -164,4 +164,20 @@ bool detail::BackendHandleVal::DoUnserializeData(BrokerDataView) {
     return false;
 }
 
+namespace detail {
+
+zeek::expected<storage::detail::BackendHandleVal*, OperationResult> BackendHandleVal::CastFromAny(Val* handle) {
+    auto b = static_cast<storage::detail::BackendHandleVal*>(handle);
```

Comment:
```suggestion
    auto b = dynamic_cast<storage::detail::BackendHandleVal*>(handle);
```

Reply:
You could probably also poke first if the Val's type tag is `TYPE_OPAQUE` and/or whether its type is `detail::backend_opaque`. That's probably faster than `dynamic_cast`.

Given this is an API on `BackendHandleVal` now (rather than static helpers) I find the mixing of cast and checking if it's open/connected unfortunate. It does more than just cast. But the API is in the detail namespace, so can also be cleaned up later I guess.

Reply:
Would `ConvertOpaque` be better?

Reply:
The type tag is an easy check, but the name doesn't get set.

Reply:
> The type tag is an easy check, but the name doesn't get set.

Sorry for the confusion - this compiles here and should work:

```
-    if ( handle->GetType()->Tag() != TYPE_OPAQUE )
+    if ( handle->GetType() != detail::backend_opaque )
         return zeek::unexpected<OperationResult>(
             OperationResult{ReturnCode::OPERATION_FAILED, "Invalid storage handle type"});
```


> Would ConvertOpaque be better?

I was more coming from the direction that cast/convert and *also* doing the the `IsOpen()` check might not be obvious. For a mouthful: `CastAndEnsureOpen` or maybe `EnsureStorageHandleOpen(Val *any)` that happens to return a casted version or an error.

But this is nitpickery.